---
title: "Spatial Phylogenetics Workshop"
format:
  revealjs:
    incremental: false
    css: styles.css
author: Joel Nitta
institute: <a href="https://joelnitta.com">https://joelnitta.com</a>
footer: <a href="https://joelnitta.github.io/spatial-phy-workshop">https://joelnitta.github.io/spatial-phy-workshop</a>
---

# Assembling Spatial and Genetic Data

## Basic inputs for spatial phylogenetics

:::: {.columns}

::: {.column width="50%"}
- A phylogeny
<br>
<br>
<br>
<br>
<br>
- Spatial occurrence data
:::

::: {.column width="50%"}
![](images/phylogeny.png){fig-alt="a phylogeny" height="200"}<br>
![](images/ocurrence.png){fig-alt="an occurrence map" height="200"}
:::

::::

... which are linked by taxonomic names (OTUs)

## Sources of occurrence data

:::: {.columns}

::: {.column width="50%"}
- Herbaria or museums
- Floras or checklists
- Previous studies
- Your own data
:::

::: {.column width="50%"}
![](images/herb_specimen.png){fig-alt="Photo of an herbarium specimen" height="450"}
:::

::::

## Online sources of occurrence data

:::: {.columns}

::: {.column width="50%"}
- GBIF
- Kew Plants of the World Online (POWO)
- VertNet

etc...
:::

::: {.column width="50%"}
![](images/gbif_logo.png){fig-alt="gbif logo" height="150"}<br>
![](images/vertnet_logo.png){fig-alt="vertnet logo" height="150"}<br>
![](images/powo_logo.png){fig-alt="powo logo" height="150"}
:::

::::

## Types of occurrence data

:::: {.columns}

::: {.column width="50%"}

Can take many forms:

- geometric shapes
- points
- checklists
- your own surveys
:::

::: {.column width="50%"}
![](https://miro.medium.com/v2/resize:fit:1284/1*TaID5vtnzOYkkKSb-VQNpg.png){fig-alt="image of points, lines, and polygons" width="550"}
:::

::::

We will focus on **point data** (the data available from GBIF) during the coding session

## GBIF <https://www.gbif.org/>

![](images/gbif.png){fig-alt="screenshot of GBIF entry page" height="450"}

## GBIF

- GBIF is not one database; it is a portal to many databases

- You should try the web interface first to familiarize yourself with it

- **We can't use the occurrence records in GBIF as-is.** They may include many errors (typos, etc.) and need to be checked carefully. More on this during the coding session.

## Cleaning occurrence data

- The settings for data cleaning **depend on your analysis**

- Defaults are a good start, but make sure they make sense!

- For example, if your grid-cell size is 1 degree x 1 degree, you don't need data to be more exact than that

## Coordinate Reference Systems (CRS) {.smaller}

:::: {.columns}

::: {.column width="50%"}
Geographic Coordinate System (**GCS**)

- *Where* the data are located
- Round (like the earth)
- Usually in degrees

Projected Coordinate System (**PCS**)

- *How* to draw a map of the data
- Flat (like a piece of paper)
- Usually in meters
:::

::: {.column width="50%"}
![](https://www.esri.com/arcgis-blog/wp-content/uploads/2022/02/grid2.png)
:::

::::

## Geographic Coordinate System (GCS)

:::: {.columns}

::: {.column width="50%"}
- Latitude and longitude alone are not enough

- The earth is not a perfect sphere

- GCS defines **how to model the earth** (e.g., WGS84)
:::

::: {.column width="50%"}

![](https://www.esri.com/arcgis-blog/wp-content/uploads/2020/02/4.png)

Hiker's coordinates at 134.577°E, 24.006°S. But where is she (A or B)?

:::

::::

::: {.medsmall}
https://www.esri.com/arcgis-blog/products/arcgis-pro/mapping/gcs_vs_pcs/
:::

## Projected Coordinate System (PCS)

:::: {.columns}

::: {.column width="50%"}
- The earth is round, but we project it onto flat maps

- The decision of how to do this is not trivial

- **There will always be some amount of distortion in area, distance, or direction**
:::

::: {.column width="50%"}
![](images/projections.png){fig-alt="Comparison of different types of georeference systems"}

::: {.small}
https://geoawesomeness.com/wp-content/uploads/2022/03/projections.jpg
:::

:::

::::

## Coordinate Reference Systems

:::: {.columns}

::: {.column width="50%"}
- You need to choose an appropriate CRS for your study (there are thousands)

- If you assume that your sampling units have equal area, make sure to use an **equal-area projection** (e.g., Mollweide)
:::

::: {.column width="50%"}
![](https://upload.wikimedia.org/wikipedia/commons/thumb/3/3f/Mollweide_with_Tissot%27s_Indicatrices_of_Distortion.svg/2560px-Mollweide_with_Tissot%27s_Indicatrices_of_Distortion.svg.png){fig-alt="The Mollweide projection with Tissot's indicatrix of deformation"}

::: {.medsmall}
The Mollweide projection. The orange dots have the same area, but their shape is distorted as you move away from the equator.
:::

::: {.small}
https://en.wikipedia.org/wiki/Mollweide_projection
:::


:::

::::

## From occurrences to grid-cells

:::: {.columns}

::: {.column width="50%"}
- Raw data are often provided on a per-species basis

- But we are interested in **assemblages** (grid-cells) of species → need to group species together
:::

::: {.column width="50%"}
![](https://scx1.b-cdn.net/csz/news/800a/2018/localconditi.jpg){fig-alt="Image of plant community"}

::: {.small}
https://phys.org/news/2018-12-local-conditions.html
:::

:::

::::

## From occurrences to grid-cells

- For point data, a typical method is to divide the study area into equal size grid-cells, then count the species occurring in each grid-cell

- For shape data, you would overlay the shapes

- For checklist data, the areas may not be equal sized. You could simply use the sampling units in the checklist (e.g., counties, countries, etc.)

## Phylogenetic data

:::: {.columns}

::: {.column width="25%"}
- Is there a tree available, or do you need to build it from scratch?
:::

::: {.column width="75%"}
![](https://images.unsplash.com/photo-1541888946425-d81bb19240f5?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=2670&q=80){fig-alt="constuction workers"}
:::

::: {.small}
https://images.unsplash.com/
:::

::::

## Sources of DNA sequence data

:::: {.columns}
::: {.column width="50%"}
- **GenBank**
- access in R via the `rentrez` package
  - (or `restez` package for larger datasets)
:::

::: {.column width="50%"}
![](https://a-little-book-of-r-for-bioinformatics.readthedocs.io/en/latest/_images/P3_image2.png){fig-alt="screenshot of genbank interface" width="450"}
:::

::: {.small}
https://a-little-book-of-r-for-bioinformatics.readthedocs.io/
:::

::::

## Building a tree from scratch

:::: {.columns}

::: {.column width="50%"}
We don't have time to cover this today - that is a whole topic of study unto itself!
:::

::: {.column width="50%"}
![](https://mediacdn.nhbs.com/jackets/jackets_resizer_xlarge/17/170234.jpg){fig-alt="Phylogenetic textbook" height="450"}
:::

::: {.small}
https://mediacdn.nhbs.com/jackets/jackets_resizer_xlarge/17/170234.jpg
:::

::::

## Sources of phylogenetic trees

:::: {.columns}

::: {.column width="50%"}
- Previous publications
- R packages that provide trees (`ftolr` for ferns)
- Open Tree of Life (`rotl` R package) (**caution!**)
- Software that places tips on the tree by taxonomy (**caution!**)
:::

::: {.column width="50%"}
![](https://docs.ropensci.org/rotl/logo.svg){fig-alt="rotl logo" width="350"}

::: {.small}
https://docs.ropensci.org/rotl/logo.svg
:::

:::

::::


## What if I don't have a tree for my group?

:::: {.columns}

::: {.column width="50%"}
- A tree at the species level may not be necessary. Consider doing the analysis at a higher taxonomic level (e.g., genus)
:::

::: {.column width="50%"}
![](https://2.bp.blogspot.com/-jbCa-ZpKCgg/VcMlTfbF-WI/AAAAAAAAwZo/7beUnZPb_MI/s800/fukidashi4_doctorw.png){fig-alt="irasuto-ya scientist thinking" width="450"}

::: {.small}
https://www.irasutoya.com/
:::

:::

::::

## Is it OK to use taxonomy in place of DNA?

In other words, to place species on the tree based on their taxonomy

- Not such a good idea. This makes a lot of assumptions.
  - Monophyly of the taxa involved
  - Correctness of the taxonomy
- Also produces weird trees with lots of polytomies that may not be suitable for spatial phylogenetics

## Taxonomic issues

- Old names
- Misspelled names
- **Mismatching synonyms**

![](https://github.com/joelnitta/biodigi_2022/blob/main/images/synonyms.png?raw=true){fig-alt="Comparison of synonyms for Crepidomanes minutum"}

We need to **resolve names** to a standard taxonomic database

# R tools for assembling spatial and genetic data

## Live coding

- Live coding session demonstrating how to use `rgbif`, `CoordinateCleaner`, and `phyloregion` to obtain data

- Code is available here: <https://github.com/joelnitta/spatial-phy-workshop/blob/main/tutorials/occ_phy.md>

## Workflow

A typical workflow involves the following steps:

1. Download occurrence records (`rgbif`)
2. Clean occurrence records (`CoordinateCleaner`)
3. Standardize species names (`rgbif`)
4. Convert points to assemblages (`phyloregion`)
5. Join assemblage data to phylogeny during spatial phylogenetic analysis (`canaper`)

## What is not covered today

- We don't have time for demonstrating how to download sequences, assemble them, and conduct phylogenetic analysis (phylogenetic pipelines).

- We will be using a pre-built tree

## Other packages: taxonomy{.smaller}

- [RTNRS](https://github.com/EnquistLab/RTNRS)
  - Standardize names using TNRS ([Taxonomic Name Resolution Service](https://tnrs.biendata.org/))
- [rWCVP](https://github.com/matildabrown/rWCVP/)
  - Standardize names to WCVP ([World Checklist of Vascular Plants](https://powo.science.kew.org/about-wcvp))
- [taxadb](https://github.com/ropensci/taxadb)
  - Standardize names using local databases
- [taxastand](https://github.com/joelnitta/taxastand)
  - Standardize names to a custom database

## Other packages: occurrences

- [occCite](https://github.com/ropensci/occCite)
  - Download occurrence data from GBIF and BIEN, and generate references for all databases that contributed
- [RBIEN](https://github.com/bmaitner/RBIEN)
  - Interface to [BIEN](https://bien.nceas.ucsb.edu/bien/) database of plant occurrences and traits

## Other packages: GenBank data

- [rentrez](https://github.com/ropensci/rentrez)
  - Interface to [Entrez](https://www.ncbi.nlm.nih.gov/search/) databases (including [GenBank](https://www.ncbi.nlm.nih.gov/genbank/))
- [restez](https://github.com/ropensci/restez)
  - Make a local copy of a portion of GenBank (good for building large trees)

## Other packages: phylogenetic pipelines

- [phruta](https://github.com/ropensci/phruta/)
  - Pipeline to download sequences, align them, and build a phylogenetic tree
- [phylotaR](https://github.com/ropensci/phylotaR)
  - Pipeline to download and cluster sequences 

## Other packages: pre-built phylogenies

- [ftolr](https://github.com/fernphy/ftolr)
  - Global fern phylogeny
- [fishtree](https://github.com/jonchang/fishtree)
  - Global fish phylogeny
- [rotl](https://github.com/ropensci/rotl)
  - R interface to the Open Tree of Life (**use with caution**)
- [U.PhyloMaker](https://github.com/jinyizju/U.PhyloMaker)
  - Assemble phylogenies by grafting species names onto a backbone (**not recommended**)

# Applications of Spatial Phylogenetics

## Spatial phylogenetics can tell us about

- The **distribution** of endemicity

- Environmental **drivers** of biodiversity

- **Structure** of biodiversity

## Distribution of endemicity

:::: {.columns}
::: {.column width="50%"}
- Paleoendemism
  - **Refugia**
  - Colonization by distantly related lineages
- Neoendemism
  - Recent speciation
- Mixed endemism
  - Multiple processes
:::


::: {.column width="50%"}
![](https://www.pnas.org/cms/10.1073/pnas.1403594111/asset/01fe3733-197f-4c31-9d83-17a440db3575/assets/graphic/pnas.1403594111fig01.jpeg){fig-alt="diagram of refugia" width="450"}

::: {.small}
<https://doi.org/10.1073/pnas.1403594111>
:::

:::

::::

## Distribution of endemicity

:::: {.columns}
::: {.column width="50%"}
- Paleoendemism
  - Refugia
  - Colonization by distantly related lineages
- Neoendemism
  - **Recent speciation**
- Mixed endemism
  - Multiple processes
:::


::: {.column width="50%"}
![](https://handwiki.org/wiki/images/9/97/Darwin%27s_finches.jpeg){fig-alt="Darwin's finches" width="450"}

::: {.small}
<https://handwiki.org/wiki/Biology:Neoendemism>
:::

:::

::::

## Case study: Ferns of Japan

:::: {.columns}

::: {.column width="50%"}
- \> 600 species
- Dense sampling
  - 10 x 10 km maps of every species
  - DNA (*rbcL*) for > 98% of species

::: {.medsmall}
Nitta et al. AJB 2022 <https://doi.org/10.1002/ajb2.1848>
:::

::: {.small}
Photos A. Ebihara
:::

:::

::: {.column width="10%"}
:::

::: {.column width="40%"}
![](https://joelnitta.github.io/botany2020/images/ja_example_ferns.png){fig-alt="alt-text" height="500"}
:::

::::

## Case study: Ferns of Japan

:::: {.columns}

::: {.column width="40%"}

- Variation in climate from N (subarctic) to S (subtropical)

- Variation in elevation

- Main islands continental, southern islands oceanic

::: {.medsmall}
Nitta et al. AJB 2022 <https://doi.org/10.1002/ajb2.1848>
:::

:::

::: {.column width="60%"}
![](https://joelnitta.github.io/botany2020/images/japan_intro.png){fig-alt="alt-text" height="550"}
:::

::::

## Skewed distribution of endemism 

:::: {.columns}
::: {.column width="50%"}
- Southern-most islands are subtropical
  - Very different climate from rest of country
- High rates of mixed- and paleo-endemism
  - **Due to distantly related (tropical) lineages**
:::

::: {.column width="50%"}
![](images/japan_endem.png){fig-alt="Pattern of phylogenetic endemism in Japan" width="450"}
:::

::::

## Reproductive mode as driver of biodiversity

:::: {.columns}

::: {.column width="50%"}
Phylogenetic diversity is predicted by % of apogamous (asexual) species

- Apogamous species tend to be **hybrids that share identical plastid sequences** with other species
:::

::: {.column width="50%"}
![](images/japan_apo.png){fig-alt="Chart showing relationship between % apogamy and PD in Japanese ferns, with negative trend" width="450"}
:::

::::

## Drivers of biodiversity

<br>

When testing **spatial** hypotheses (e.g., richness is determined by temperature), we must use **spatial** methods

. . . 

<br>

::: {.medlarge}
Because of **spatial autocorrelation**
:::

## Spatial autocorrelation

![](https://mgimond.github.io/Spatial/img/Random_maps.png){fig-alt="Hypothetical maps of US showing effect of spatial autocorrelation" height="550"}

::: {.small}
<https://mgimond.github.io/Spatial/spatial-autocorrelation.html>
:::

## Accounting for spatial autocorrelation

Compare amount of observed autocorrelation to some expected value: **Moran's _I_**

![](https://static.cambridge.org/binary/version/id/urn:cambridge.org:id:binary:20200421144010724-0711:9781108614528:49898fig4_1.png?pub-status=live){fig-alt="Image of different values of Moran's I" height="400"}

::: {.small}
<https://www.cambridge.org/core/books/abs/spatial-analysis-methods-and-practice/spatial-autocorrelation/F6A01B574C69076F28318445C33397E4>
:::

## Accounting for spatial autocorrelation

:::: {.columns}

::: {.column width="50%"}
Workflow:

- Conduct non-spatial analysis
- Check degree of Moran's *I* in model residuals
- If significant, re-do analysis using spatial model
:::

::: {.column width="50%"}
![](https://anddavies.co.uk/wp-content/uploads/2020/02/rspb.2020.287.issue-1920.largecover-1024x675.jpg){fig-alt="Image showing river elevation in relief" width="450"}
:::

::::

## Structure of biodiversity

Understanding the distribution of bio-regions

- Used to be done *ad-hoc*
  - Not objective

- Newer approaches are quantitative (covered by Brent's talk)

## 3-4 regions in ferns of Japan

![](images/japan_region.png){fig-alt="Figure showing bioregions of ferns of Japan" height="450"}

High rates of endemism on remote islands cause difference in taxonomic and phylogenetic bioregions

## Rates of protection vary by biodiversity metric

![](images/japan_protected.png){fig-alt="Figure showing bioregions of ferns of Japan" height="450"}

# Spatial phylogenetics in R with `canaper`

## Live coding

- Live coding session demonstrating how to use `canaper` to conduct spatial phylogentic analysis

- Code is available here: <https://github.com/joelnitta/spatial-phy-workshop/blob/main/tutorials/canaper.md>

## Workflow

A typical workflow involves the following steps:

1. Load phylogeny and grid-cell data
2. Determine appropriate null model
3. Conduct randomization test
4. Categorize results
5. Plot results

## Survey

Please fill out the post-workshops survey:

<https://forms.gle/Dh1TdJctskWXVHSW7>
